packages/rs-sdk/src/core/transaction.rs

```diff
@@ -135,58 +133,140 @@ impl Sdk {
+                                    },
+                                    RequestSettings::default(),
+                                )
+                                .await?;
+
+                            core_chain_locked_height = height;
+
+                            if is_chain_locked {
+                                break;
+                            }
+
+                            tracing::trace!("the transaction is on height {} but not chainlocked. try again in 1 sec", height);
+
+                            sleep(Duration::from_secs(1)).await;
+                        }
+
+                        tracing::debug!(
+                            "the transaction is chainlocked on height {}, waiting platform for reaching the same core height",
+                            core_chain_locked_height
+                        );
+
+                        // Wait until platform chain is on the block's chain locked height
+                        loop {
+                            let request = GetEpochsInfoRequest {
+                                version: Some(get_epochs_info_request::Version::V0(
+                                    get_epochs_info_request::GetEpochsInfoRequestV0 {
+                                        start_epoch: Some(0),
+                                        count: 1,
+                                        ..Default::default()
+                                    },
+                                )),
+                            };
 
-                        continue;
+                            let GetEpochsInfoResponse {
+                                version:
+                                    Some(get_epochs_info_response::Version::V0(
+                                        get_epochs_info_response::GetEpochsInfoResponseV0 {
+                                            metadata: Some(metadata),
+                                            ..
+                                        },
+                                    )),
+                            } = self.execute(request, RequestSettings::default()).await?
+                            else {
+                                return Err(Error::DapiClientError(String::from(
+                                    "missing V0 `metadata` field",
+                                )));
+                            };
+
+                            if metadata.core_chain_locked_height >= core_chain_locked_height {
+                                break;
+                            }
+
+                            tracing::trace!(
+                                "platform chain locked core height {} but we need {}. try again in 1 sec",
+                                metadata.core_chain_locked_height,
+                                core_chain_locked_height,
+                            );
+
+                            sleep(Duration::from_secs(1)).await;
+                        }
```

Comment:
I feel like there's something that can be done much better than this.

Reply:
Yeah, the better solution will be when we run header and transaction streams in background threads so we always know what is the height and chainlocks atm how it's done in JS wallet. But I'm not sure we want to invest time to this right now.